lib/plugins.ts

```diff
@@ -0,0 +1,18 @@
+import { IGenericModule, IPlugin, IState } from './interfaces'
+
+export const LocalStoragePlugin = <S extends IState>(
```

Comment:
Generic localstorage plugin. It would be nice to give plugins a way to operate on their children, but I'm not sure the best interface for doing that.

On the other hand, at least for the specific case of storing state in localstorage, maybe it's better not to nest the stored state anyway. It will be more performant, and only one store needs to be cleared when the version is updated. I worry that people would expect the storage data layout to match the module/submodules, though.